Error when reading tfevents files written with TensorBoardLogger v0.1.21

Hello, I updated `TensorboardLogger` to the latest `0.1.21` version, and the following minimum example doesn't work anymore:

```julia
using Logging
using TensorBoardLogger
using ValueHistories

logger = TBLogger("logs"; min_level=Logging.Info)

with_logger(logger) do
    @info "train" loss = 1.0
    @info "train" loss = 0.0
end

r = TBReader("logs1")
hist = MVHistory()

TensorBoardLogger.map_summaries(r) do tag, iter, val
    push!(hist, Symbol(tag), iter, val)
end
```

Here is the error:

```bash
ERROR: AssertionError: crc_data == crc_data_ck
Stacktrace:
 [1] read_event(f::IOStream)
   @ TensorBoardLogger ~/.julia/packages/TensorBoardLogger/LO5eP/src/Deserialization/deserialization.jl:49
 [2] iterate(it::TensorBoardLogger.TBEventFileIterator, state::Int64)
   @ TensorBoardLogger ~/.julia/packages/TensorBoardLogger/LO5eP/src/Deserialization/deserialization.jl:126
 [3] map_summaries(fun::var"#21#22", logdir::TBReader; purge::Bool, tags::Nothing, steps::Nothing, smart::Bool)
   @ TensorBoardLogger ~/.julia/packages/TensorBoardLogger/LO5eP/src/Deserialization/deserialization.jl:270
 [4] map_summaries(fun::Function, logdir::TBReader)
   @ TensorBoardLogger ~/.julia/packages/TensorBoardLogger/LO5eP/src/Deserialization/deserialization.jl:248
```

Reading files written with an older version still works.
Am I missing something ?

ps: by the way, it seems that version `0.1.20` cannot be installed from the general registry for some reason (only from the github repository url)

--- a/Project.toml
+++ b/Project.toml
@@ -1,7 +1,7 @@
 name = "TensorBoardLogger"
 uuid = "899adc3e-224a-11e9-021f-63837185c80f"
 authors = ["Filippo Vicentini <filippovicentini@gmail.com>"]
-version = "0.1.21"
+version = "0.1.22"
 
 [deps]
 CRC32c = "8bf52ea8-c179-5cab-976a-9e18b702a9bc"

--- a/src/event.jl
+++ b/src/event.jl
@@ -29,17 +29,16 @@ format. The format follows the following rule (in bytes)
 #4 N..N+8      UInt32  masked_CRC of #3
 """
 function write_event(out::IO, event::Event)
-    data = PipeBuffer();
-    encode(ProtoEncoder(data), event)
+    event_bytes = serialize_proto(event)
 
     #header
-    header     = collect(reinterpret(UInt8, [data.size]))
+    header     = collect(reinterpret(UInt8, [length(event_bytes)]))
     crc_header = reinterpret(UInt8, UInt32[masked_crc32c(header)])
-    crc_data   = reinterpret(UInt8, UInt32[masked_crc32c(data.data)])
+    crc_data   = reinterpret(UInt8, UInt32[masked_crc32c(event_bytes)])
 
     write(out, header)
     write(out, crc_header)
-    write(out, data.data)
+    write(out, event_bytes)
     write(out, crc_data)
     flush(out)
 end

--- a/src/hparams.jl
+++ b/src/hparams.jl
@@ -73,12 +73,6 @@ function metric_info(c::MetricConfig)
     return HP.MetricInfo(mname, c.displayname, c.description, HDatasetType.DATASET_UNKNOWN)
 end
 
-function encode_bytes(content::HP.HParamsPluginData)
-    data = PipeBuffer()
-    encode(ProtoEncoder(data), content)
-    return take!(data)
-end
-
 # Dictionary serialisation in ProtoBuf does not work for this specific map type
 # and must be overloaded so that it can be parsed. The format was derived by
 # looking at the binary output of a log file created by tensorboardX.
@@ -156,17 +150,17 @@ function write_hparams!(logger::TBLogger, hparams::Dict{String,<:Any}, metrics::
 
     experiment = HP.Experiment("", "", "", time(), hparam_infos, metric_infos)
     experiment_content = HP.HParamsPluginData(PLUGIN_DATA_VERSION, OneOf(:experiment, experiment))
-    experiment_md = SummaryMetadata(SummaryMetadata_PluginData(PLUGIN_NAME, encode_bytes(experiment_content)), "", "", DataClass.DATA_CLASS_UNKNOWN)
+    experiment_md = SummaryMetadata(SummaryMetadata_PluginData(PLUGIN_NAME, serialize_proto(experiment_content)), "", "", DataClass.DATA_CLASS_UNKNOWN)
     experiment_summary = Summary([Summary_Value("", EXPERIMENT_TAG, experiment_md, nothing)])
 
     session_start_info = HP.SessionStartInfo(hparams_dict, "", "", "", time())
     session_start_content = HP.HParamsPluginData(PLUGIN_DATA_VERSION, OneOf(:session_start_info, session_start_info))
-    session_start_md = SummaryMetadata(SummaryMetadata_PluginData(PLUGIN_NAME, encode_bytes(session_start_content)), "", "", DataClass.DATA_CLASS_UNKNOWN)
+    session_start_md = SummaryMetadata(SummaryMetadata_PluginData(PLUGIN_NAME, serialize_proto(session_start_content)), "", "", DataClass.DATA_CLASS_UNKNOWN)
     session_start_summary = Summary([Summary_Value("", SESSION_START_INFO_TAG, session_start_md, nothing)])
 
     session_end_info = HP.SessionEndInfo(HP.Status.STATUS_SUCCESS, time())
     session_end_content = HP.HParamsPluginData(PLUGIN_DATA_VERSION, OneOf(:session_end_info, session_end_info))
-    session_end_md = SummaryMetadata(SummaryMetadata_PluginData(PLUGIN_NAME, encode_bytes(session_end_content)), "", "", DataClass.DATA_CLASS_UNKNOWN)
+    session_end_md = SummaryMetadata(SummaryMetadata_PluginData(PLUGIN_NAME, serialize_proto(session_end_content)), "", "", DataClass.DATA_CLASS_UNKNOWN)
     session_end_summary = Summary([Summary_Value("", SESSION_END_INFO_TAG, session_end_md, nothing)])
 
     for s in (experiment_summary, session_start_summary, session_end_summary)

--- a/src/utils.jl
+++ b/src/utils.jl
@@ -9,7 +9,7 @@ end
 function serialize_proto(data)
     pb = PipeBuffer()
     encode(ProtoEncoder(pb), data)
-    return view(pb.data, 1:pb.size) # Do not return additional zeros
+    return take!(pb) # Do not return additional zeros
 end
 
 """

--- a/src/utils.jl
+++ b/src/utils.jl
@@ -9,7 +9,7 @@ end
 function serialize_proto(data)
     pb = PipeBuffer()
     encode(ProtoEncoder(pb), data)
-    pb.data
+    return view(pb.data, 1:pb.size) # Do not return additional zeros
 end
 
 """

--- a/test/test_hparams.jl
+++ b/test/test_hparams.jl
@@ -65,7 +65,7 @@ const hparams_metadata_encoded_bytes = UInt8[0x1a, 0x5b, 0x0a, 0x0e, 0x0a, 0x06,
     session_start_info = TensorBoardLogger.HP.SessionStartInfo(hparams_dict, "", "", "", zero(Float64))
     session_start_content = TensorBoardLogger.HP.HParamsPluginData(PLUGIN_DATA_VERSION, TensorBoardLogger.OneOf(:session_start_info, session_start_info))
     
-    @test TensorBoardLogger.encode_bytes(session_start_content) == hparams_metadata_encoded_bytes
+    @test TensorBoardLogger.serialize_proto(session_start_content) == hparams_metadata_encoded_bytes
 end
 
 

